syntax = "proto3";

// Package for file checksum services.
package checksum;


// Service that handles file uploads and computes file checksums.
service FileChecksumService
{
	// RPC method for uploading a file.
	// The client sends a FileUploadRequest, and the server returns the checksum in a FileUploadResponse.
	rpc UploadFile(FileUploadRequest) returns (FileUploadResponse);
}

// Message containing the file data for upload.
message FileUploadRequest
{
	// [required] The file data to be uploaded as bytes.
	bytes file_data = 1;
}

// Message containing the computed checksum of the uploaded file.
message FileUploadResponse 
{
	// [required] The computed checksum of the uploaded file as a string.
	string checksum = 1;
}
